.github/actions/pylint/index.js

```diff
@@ -0,0 +1,137 @@
+            }
+        });
+    } catch (pylintError) {
+        try {
+            reportResults(JSON.parse(pylintOutput), pylintDisable);
+        } catch (reportingError) {
+            // The exception thrown by `exec` doesn't include the exit
+            // code as a field (we'd have to parse the error message).
+            // So report the pylint process error here as well, in case
+            // that was the root cause for `reportResults` failing.
+            console.error(pylintError);
+            console.error(reportingError);
+        }
+    }
+}
+
+/**
+ * Create GitHub annotations and set the action to failed.
+ *
+ * Report a summary followed by per-line annotations.
+ *
+ * @param pylintMessages List of Pylint messages decoded from JSON.
+ * @param pylintDisable Disabled checks to add to summary, potentially
+ *  useful for debugging.
+ *
+ */
+function reportResults(pylintMessages, pylintDisable) {
+    const pylintErrors = pylintMessages.filter(message => message.type === 'error');
+    const pylintWarnings = pylintMessages.filter(message => message.type === 'warning');
+    const pylintConvention = pylintMessages.filter(message => message.type === 'convention');
+    const pylintRefactor = pylintMessages.filter(message => message.type === 'refactor');
+    const pylintInfo = pylintMessages.filter(message => message.type === 'info');
+
+    // Construct summary report.
+
+    let summaryTitle = "Pylint linter issues detected";
+    if (pylintDisable) {
+        summaryTitle += ` (disabled checks '${pylintDisable}')`;
+    }
+    const summaryLines = [summaryTitle];
+
+    const appendToSummaryIfNonEmpty = (messages, severity) => messages.length &&
+        summaryLines.push(`${messages.length} ${severity}`);
+
+    appendToSummaryIfNonEmpty(pylintErrors, "error");
+    appendToSummaryIfNonEmpty(pylintWarnings, "warning");
+    appendToSummaryIfNonEmpty(pylintConvention, "convention");
+    appendToSummaryIfNonEmpty(pylintRefactor, "refactor");
+    appendToSummaryIfNonEmpty(pylintInfo, "info");
+
+    core.setFailed(summaryLines.join("\n\n"));
+
+    // Annotate individual issues.
+
+    pylintErrors.forEach(
+        message => core.error(
+            annotationMessage(message),
+            annotationProperties(message)));
+
+    [...pylintWarnings, ...pylintRefactor, ...pylintConvention].forEach(
+        message => core.warning(
+            annotationMessage(message),
+            annotationProperties(message)));
+
+    pylintInfo.forEach(
+        message => core.notice(
+            annotationMessage(message),
+            annotationProperties(message)));
+}
+
+/**
+ * Parse a Pylint message into GitHub annotation properties.
+ *
+ * @param message Pylint message decoded from JSON.
+ * @returns core.AnnotationProperties Github annotation properties.
+ */
+function annotationProperties(message) {
+    return {
+        title: `Linter ${message.type} ${message['message-id']} (${message.symbol})`,
```

Comment:
[question] Annoyingly, the annotation title doesn't show up in the action log. The title only seems to show when displayed in the action "Summary" or when displayed alongside the source.

Do we care?

Reply:
The appearance as-is is still pretty good - and a million times better than the previous state. So I'm happy as is.